sdk/metric/metrictest/exporter.go

```diff
@@ -0,0 +1,174 @@
+	}
+
+	return c, exp
+}
+
+// ExportRecord represents one collected datapoint from the Exporter.
+type ExportRecord struct {
+	InstrumentName         string
+	InstrumentationLibrary Library
+	Labels                 []attribute.KeyValue
+	AggregationKind        aggregation.Kind
+	Sum                    number.Number
+	Count                  uint64
+	Histogram              aggregation.Buckets
+	LastValue              number.Number
+}
+
+// Collect triggers the SDK's collect methods and then aggregates the data into
+// `ExportRecord`s.
+func (e *Exporter) Collect(ctx context.Context) error {
+	e.exports = []ExportRecord{}
+
+	err := e.controller.Collect(ctx)
+	if err != nil {
+		return err
+	}
+
+	return e.controller.ForEach(func(l instrumentation.Library, r export.Reader) error {
+		lib := Library{
+			InstrumentationName:    l.Name,
+			InstrumentationVersion: l.Version,
+			SchemaURL:              l.SchemaURL,
+		}
+
+		return r.ForEach(aggregation.CumulativeTemporalitySelector(), func(rec export.Record) error {
+			record := ExportRecord{
+				InstrumentName:         rec.Descriptor().Name(),
+				InstrumentationLibrary: lib,
+				Labels:                 rec.Labels().ToSlice(),
+				AggregationKind:        rec.Aggregation().Kind(),
+			}
+
+			var err error
+			switch agg := rec.Aggregation().(type) {
+			case aggregation.Histogram:
+				record.AggregationKind = aggregation.HistogramKind
+				record.Histogram, err = agg.Histogram()
+				if err != nil {
+					return err
+				}
+				record.Sum, err = agg.Sum()
+				if err != nil {
+					return err
+				}
+				record.Count, err = agg.Count()
+				if err != nil {
+					return err
+				}
+			case aggregation.Count:
+				record.Count, err = agg.Count()
+				if err != nil {
+					return err
+				}
+			case aggregation.LastValue:
+				record.LastValue, _, err = agg.LastValue()
+				if err != nil {
+					return err
+				}
+			case aggregation.Sum:
+				record.Sum, err = agg.Sum()
+				if err != nil {
+					return err
+				}
+			}
+
+			e.exports = append(e.exports, record)
+			return nil
+		})
+	})
+}
+
+// GetRecords returns all Records found by the SDK
+func (e *Exporter) GetRecords() []ExportRecord {
+	return e.exports
+}
+
+var ErrNotFound = fmt.Errorf("record not found")
+
+// GetByName returns the first Record with a matching name.
+func (e *Exporter) GetByName(name string) (ExportRecord, error) {
+	for _, rec := range e.exports {
+		if rec.InstrumentName == name {
+			return rec, nil
+		}
+	}
+	return ExportRecord{}, ErrNotFound
+}
+
+// GetByNameAndLabels returns the first Record with a matching name and set of labels
+func (e *Exporter) GetByNameAndLabels(name string, labels []attribute.KeyValue) (ExportRecord, error) {
+	for _, rec := range e.exports {
+		if rec.InstrumentName == name && labelsMatch(labels, rec.Labels) {
+			return rec, nil
+		}
+	}
+	return ExportRecord{}, ErrNotFound
+}
+
+func labelsMatch(labelsA, labelsB []attribute.KeyValue) bool {
+	if len(labelsA) != len(labelsB) {
```

Comment:
It might be worth using `Set.Equal` here instead

https://github.com/open-telemetry/opentelemetry-go/blob/f6d691efa75a9081db0c288a1ae770cef58a4ef8/attribute/set.go#L172-L175

That way different attribute order does not cause this to fail.

Reply:
Really what we would want here is if B is a subset of A.  That way we don't have to specify ALL the labels, just the ones we are concerned with.

Reply:
Hmm, okay. It looks like the algorithm might need to be revised if that is what we are trying to do here. Maybe constructing map of distinct attribute values and ensure they all exist in the other.

Reply:
The first pass of this was just if they were equal, but the more useful notion is to match a subset.  I found an easy way of doing this.